apps/tenant-management-webapp-e2e/src/integration/events/events.step.ts

```diff
@@ -175,3 +175,257 @@ Then('the user views the details of {string} under Core streams', function (stre
+          Cypress.log({
+            name: 'Row number for the found file type: ',
+            message: String(rowNumber),
+          });
+          resolve(rowNumber);
+        });
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+Then('the user {string} the stream of {string}, {string}', function (viewOrNot, streamName, role) {
+  findStream(streamName, role).then((rowNumber) => {
+    switch (viewOrNot) {
+      case 'views':
+        expect(rowNumber).to.be.greaterThan(0, 'Stream of ' + streamName + ', ' + role + ' has row #' + rowNumber);
+        break;
+      case 'should not view':
+        expect(rowNumber).to.equal(0, 'Stream of ' + streamName + ', ' + role + ' has row #' + rowNumber);
+        break;
+      default:
+        expect(viewOrNot).to.be.oneOf(['views', 'should not view']);
+    }
+  });
+});
+
+Then(
+  'the user views the stream details of {string}, {string}, {string}, {string}',
+  function (streamName, description, event, role) {
+    eventsObj.streamDetails(streamName).should('contain', streamName);
+    eventsObj.streamDetails(streamName).should('contain', description);
+    if (role == 'public') eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": true');
+    else {
+      eventsObj.streamDetails(streamName).should('contain', '"publicSubscribe": false');
+    }
+    const roles = role.split(',');
+    for (let i = 0; i < roles.length; i++) {
+      eventsObj
+        .streamDetails(streamName)
+        .invoke('text')
+        .then((roleDetails) => {
+          const subscriberRoles = roles[i].split(',')[0].trim();
+          expect(roleDetails).to.contain(subscriberRoles);
+        });
+    }
+    const events = event.split(',');
+    for (let i = 0; i < events.length; i++) {
+      eventsObj
+        .streamDetails(streamName)
+        .invoke('text')
+        .then((eventDetails) => {
+          const namespace = events[i].split(':')[0].trim();
+          const name = events[i].split(':')[1].trim();
+          expect(eventDetails).to.contain('"namespace": ' + '"' + namespace + '"');
+          expect(eventDetails).to.contain('"name": ' + '"' + name + '"');
+        });
+    }
+  }
+);
+
+When('the user clicks {string} button of {string}', function (button, streamName) {
+  switch (button) {
+    case 'Eye':
+      eventsObj.streamDetailsEyeIcon(streamName).click();
+      break;
+    case 'Eye-Off':
+      eventsObj.streamDetailsEyeOffIcon(streamName).click();
+      break;
+    case 'Edit':
+      eventsObj.streamEditBtn(streamName).click();
+      break;
+    case 'Delete':
+      eventsObj.streamDeleteBtn(streamName).click();
+      break;
+    default:
+      expect(button).to.be.oneOf(['Eye', 'Edit', 'Delete']);
+  }
+});
+
+Then('the user views Edit stream modal', function () {
+  eventsObj.streamModalTitle().invoke('text').should('eq', 'Edit stream');
+});
+
+Then('the user enters {string}, {string}, {string} in Edit stream modal', function (description, event, role) {
```

Comment:
Need a comment about n/a so that user calling this step knows which parameter(s) allow n/a and what n/a parameter does.